### DISCUSSION TITLE
Symbolic generalization of the Wigner rotation

### DISCUSSION THREAD

**Discussion by mmikhasenko** [ComPWA/report#18] [Ideas]
The note based on the discussion today

The problem  is to find symbolic expressions for the Wigner rotation needed to align two decay chains
Example: B->Jpsi pi pi K
```
1) B->X(->Jpsi rho(->pi pi)) K, and
2) B->K**(->K*(->K pi) pi) Jpsi 
```
The expressions are not known in general. I have not seem numerical implementation.
I am not aware of the four body analysis with spin in the final state.
One needs to check what AmpGen does, Tim Evens has invested some work into it.

For the example above, the amplitude reads:
```
A = DDDH #(1)
 + DDDH D(wigner) #2
```

The algorithm to find the Wigner rotations proceeds in steps:
0. decide on what your variables are,
1. kappa2fourvectors()
- for 3b, the vectors are in the plane. Their momenta and energy are functions of m12 and m23, as well as the relative angles.
See an example of the four-vector construction in [TensorFlowAnalysis/Kinematics.py](https://gitlab.cern.ch/poluekt/TensorFlowAnalysis/-/blob/master/TensorFlowAnalysis/Kinematics.py#L364)
2. Go to the rest frame of the particle of interest (the one with a spin) in two different ways
3. Find the Euler transformation (Z-Y-Z) that relate the two spatial configuration of the rigid body of the momenta orientation.

**Comment by redeboer** [comment]
Overview of the three alignment papers here: https://github.com/ComPWA/ampform/issues/6
